modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -80,213 +139,233 @@
 		return
-	world_check += 5 SECONDS
+	COOLDOWN_START(src, forging_cooldown, 5 SECONDS)
 	check_fuel()
 	check_temp()
 	check_in_use() //plenty of weird bugs, this should hopefully fix the in_use bugs
 
+/obj/structure/reagent_forge/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	var/user_smithing_skill = user.mind.get_skill_level(/datum/skill/smithing)
+	switch(user_smithing_skill)
+		if(SKILL_LEVEL_NONE to SKILL_LEVEL_NOVICE)
+			to_chat(user, span_notice("[src] requires you to be more experienced!"))
+		if(SKILL_LEVEL_APPRENTICE to SKILL_LEVEL_JOURNEYMAN)
+			if(forge_level >= FORGE_LEVEL_ONE)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_ONE
+			goliath_ore_improvement = 3
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		if(SKILL_LEVEL_EXPERT to SKILL_LEVEL_MASTER)
+			if(forge_level >= FORGE_LEVEL_TWO)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_TWO
+			sinew_lower_chance = 100
+			current_sinew = 10
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		if(SKILL_LEVEL_LEGENDARY)
+			if(forge_level >= FORGE_LEVEL_THREE)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_THREE
+			current_core = 3
+			create_reagent_forge()
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+
 /obj/structure/reagent_forge/attackby(obj/item/I, mob/living/user, params)
 	if(istype(I, /obj/item/stack/sheet/mineral/wood)) //used for weak fuel
+		var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER) * 3 SECONDS
 		if(in_use) //only insert one at a time
 			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
 			return
 		in_use = TRUE
-		if(forge_fuel_weak >= 300) //cannot insert too much
-			to_chat(user, span_warning("You only need one to two pieces of wood at a time! You have [forge_fuel_weak] seconds remaining!"))
-			in_use = FALSE
+		if(forge_fuel_weak >= 5 MINUTES) //cannot insert too much
+			fail_message(user, "You only need one to two pieces of wood at a time! You have [forge_fuel_weak/10] seconds remaining!")
 			return
-		to_chat(user, span_warning("You start to throw the fuel into the forge..."))
-		if(!do_after(user, 3 SECONDS, target = src)) //need 3 seconds to fuel the forge
-			to_chat(user, span_warning("You abandon fueling the forge."))
-			in_use = FALSE
+		to_chat(user, span_warning("You start to throw [I] into [src]..."))
+		if(!do_after(user, skill_modifier, target = src)) //need 3 seconds to fuel the forge
```

Comment:
the comment doesnt work, now that skill_modifier is varied and not always 3 seconds;
```suggestion
		if(!do_after(user, skill_modifier, target = src)) //need to take a moment to fuel the forge
```